app/models/moderator_application.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Represents a moderator application.
+class ModeratorApplication < ApplicationRecord
+  LENGTH_RESTRICTION = {
+    maximum: 2000,
+    too_long: 'This is not the place for your life story. Limit reponses to 2000 charcters.'
+  }.freeze
+
+  belongs_to :user
+
+  # The ID of the user who is applying.
+  # Users may only have one moderator application each; however, they may edit it.
+  validates :user_id, presence: true, uniqueness: true
+
+  # The user must confirm that they are at least 18 years old.
+  # This data is not stored in the database, but the box must be checked on the form.
+  validates :at_least_18_years_old,
+            acceptance: { message: 'You must be at least 18 years old to apply to be a moderator.' }
+
+  # The user's time zone, in Ruby's TimeZone format.
+  validates :time_zone, presence: { message: 'You must specify your time zone.' }
+  validates_each :time_zone do |record, attr, value|
+    if ActiveSupport::TimeZone[value].nil?
+      record.errors.add(attr, 'You must supply a valid time zone.')
+    end
+  end
+
+  # The hours that the user would be available to actively moderate.
+  validates :active_hours,
+            presence: { message: 'You must specify when you can actively moderate the site.' },
+            length: LENGTH_RESTRICTION
```

Comment:
The formatting throughout this file is really ugly and inconsistent, but I'm not sure what to do to make it better. Putting the length before the message would help, but I think the real solution would be to fix the stupid crap I've done with the error messages in the first place.

The more I look at this file, the more I hate it. It's like a wart on my code.